logstash-core/src/main/java/org/logstash/ackedqueue/io/AbstractByteBufferPageIO.java

```diff
@@ -169,12 +175,26 @@ private void readNextElement(long expectedSeqNum, boolean verifyChecksum) throws
         }
 
         // at this point we recovered a valid element
-        this.offsetMap.add(elementOffset);
+        storeOffset(elementOffset);
         this.head = newHead + length + CHECKSUM_SIZE;
 
         buffer.position(this.head);
     }
 
+    /**
+     * Store one offset in {@link AbstractByteBufferPageIO#offsetMap}, resizing
+     * {@link AbstractByteBufferPageIO#offsetMap} if necessary.
+     * @param elementOffset Offset to store
+     */
+    private void storeOffset(final int elementOffset) {
```

Comment:
This method needs tests around its behavior for sure!

Yes, its simple, but we want it verified :)

I actually am thinking that this may be something we reuse. What are your thoughts on extracting this into a new `PrimitiveIntegerList` class?

Reply:
Also, even if we don't reuse it, it makes the test case more clear

Reply:
@andrewvc I'll extract this, sounds easy enough :)

Reply:
@andrewvc though I'd call it `IntVector`, it's not really a `List`? :)